apps/prairielearn/src/pages/instructorStudents/components/StudentsCard.tsx

```diff
@@ -0,0 +1,145 @@
+
+// This default must be declared outside the component to ensure referential
+// stability across renders, as `[] !== []` in JavaScript.
+const DEFAULT_SORT: SortingState = [];
+
+export function StudentsCard({ students }: { students: StudentRow[] }) {
+  const [globalFilter, setGlobalFilter] = useQueryState('search', parseAsString.withDefault(''));
+  const [sorting, setSorting] = useQueryState<SortingState>(
+    'sort',
+    parseAsSortingState.withDefault(DEFAULT_SORT),
+  );
+
+  const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([]);
+  const [columnSizing, setColumnSizing] = useState<ColumnSizingState>({});
+  const [columnVisibility, setColumnVisibility] = useState<ColumnVisibilityState>({});
+  const [columnPinning, setColumnPinning] = useState<ColumnPinningState>({
+    left: [],
+    right: [],
+  });
+
+  const columns = useMemo<ColumnDef<StudentRow, any>[]>(
+    () => [
+      columnHelper.accessor('uid', {
+        header: 'UID',
+        cell: (info) => info.getValue(),
+      }),
+      columnHelper.accessor('name', {
+        header: 'Name',
+        cell: (info) => info.getValue() || '—',
+      }),
+      columnHelper.accessor('email', {
+        header: 'Email',
+        cell: (info) => info.getValue() || '—',
+      }),
+      columnHelper.accessor('created_at', {
+        header: 'Enrolled At',
+        cell: (info) => {
+          const date = new Date(info.getValue());
+          return date.toLocaleDateString() + ' ' + date.toLocaleTimeString();
+        },
+      }),
+    ],
+    [],
+  );
+
+  const table = useReactTable({
+    data: students,
+    columns,
+    columnResizeMode: 'onChange',
+    getRowId: (row) => row.uid,
+    state: {
+      sorting: sorting ?? undefined,
+      columnFilters,
+      globalFilter,
+      columnSizing,
+      columnVisibility,
+      columnPinning,
+    },
+    onSortingChange: setSorting,
+    onColumnFiltersChange: setColumnFilters,
+    onGlobalFilterChange: setGlobalFilter,
+    onColumnSizingChange: setColumnSizing,
+    onColumnVisibilityChange: setColumnVisibility,
+    onColumnPinningChange: setColumnPinning,
+    getCoreRowModel: getCoreRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+    getFilteredRowModel: getFilteredRowModel(),
+    defaultColumn: {
+      minSize: 150,
+      size: 300,
+      maxSize: 500,
+      enableSorting: true,
+      enableHiding: true,
+      enablePinning: true,
+    },
+  });
+
+  return (
+    <div class="card d-flex flex-column h-100">
+      <div class="card-header bg-primary text-white">
+        <div class="d-flex align-items-center justify-content-between">
+          <div>Students</div>
+          <div>
+            <DownloadButton students={students} table={table} />
+          </div>
+        </div>
+      </div>
+      <div class="card-body d-flex flex-column">
+        <div class="d-flex flex-row mb-2">
+          <div class="col-xl-4 col-md-6 col-8 col-auto">
+            <input
```

Comment:
When using virtualized rendering, the browser search experience (Cmd/Ctrl-F) ceases to be particularly useful. A nice pattern we've used on other pages like the gradebook is to give focus to the search input when the find shortcut is pressed:

https://github.com/PrairieLearn/PrairieLearn/blob/ffc6e64cccc5c6950feabddd5d27c0e53b013093/apps/prairielearn/assets/scripts/instructorGradebookClient.ts#L167-L177

Importantly, we allow the event to bubble up to the browser if the shortcut is used twice in a row in case someone _really_ wants to use the built-in find.

Reply:
Done in 614a341082aa1f87fbfbc9e40304d4862aa9c449. It bubbles up too.